src/Microsoft.AspNetCore.Mvc.Core/ChallengeResult.cs

```diff
@@ -101,8 +101,10 @@ public ChallengeResult(IList<string> authenticationSchemes, AuthenticationProper
             var loggerFactory = context.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>();
             var logger = loggerFactory.CreateLogger<ChallengeResult>();
 
+            logger.ChallengeResultExecuting(AuthenticationSchemes);
+
             var authentication = context.HttpContext.Authentication;
-            if (AuthenticationSchemes.Count > 0)
+            if (AuthenticationSchemes != null && AuthenticationSchemes.Count > 0)
```

Comment:
I've added a null check here, since `AuthenticationSchemes` can be null (there's no null guard in the constructor).